crates/sui-source-validation/src/lib.rs

```diff
@@ -23,6 +47,10 @@ use sui_types::base_types::ObjectID;
 #[cfg(test)]
 mod tests;
 
+const CURRENT_COMPILER_VERSION: &str = env!("CARGO_PKG_VERSION");
+const LEGACY_COMPILER_VERSION: &str = CURRENT_COMPILER_VERSION; /* TODO: update this when Move 2024 is released */
```

Comment:
Note palceholder to pin the legacy compiler to the previous version when Move 2024 is enabled (current version is 1.17.0 in main)